### PR TITLE
fix: fres, sres, minres, satstd requires a field for the polynomial c…

### PR DESCRIPTION
**Pull request by hannes14**
…oeffs.

### CODE REVIEW

File: src/ideal/ideal.jl [hunk 1]
```diff
@@ -618,7 +618,7 @@ the syzygy module of the previous module, starting with the given
 ideal/module.
 The `max_length` can be set to $0$ if the full free resolution is required.
 """
-function fres(id::Union{sideal{T}, smodule{T}}, max_length::Int, method::String = "complete") where T <: Nemo.RingElem
+function fres(id::Union{sideal{T}, smodule{T}}, max_length::Int, method::String = "complete") where T <: Nemo.FieldElem
```
Comment (tthsqe12) [inline src/ideal/ideal.jl]: Unfortunately `T` here is the polynomial ring element, which is never going to be a field element.
See https://github.com/oscar-system/Singular.jl/pull/447/checks?check_run_id=2963461194#step:6:224
Something like this is probably the way to go.
```julia
function fres(id::Union{sideal{spoly{T}}, smodule{spoly{T}}}, max_length::Int, method::String = "complete") where T <: Nemo.FieldElem
   ...
   return sresolution{spoly{T}}(R, r, Bool(minimal))
end

```

**Review by tthsqe12 — CHANGES_REQUESTED** [review]

File: src/ideal/ideal.jl (hunk 1 above)
Comment (fingolfin) [reply src/ideal/ideal.jl]: @hannes14 did you see this comment?

Comment (hannes14) [reply src/ideal/ideal.jl]: Yes, I am working on it with this hint.@fingolfin, @tthsqe12

**Comment by tthsqe12** [comment]
Maybe before extending this to rings, we should make sure the existing functions are correct? #114 is still open ...

File: src/module/module.jl [hunk 2]
```diff
@@ -187,13 +187,13 @@ end
...
 @doc Markdown.doc"""
-    sres{T <: Nemo.RingElem}(I::smodule{T}, max_length::Int)
+    sres{T <: Singular.FieldElem}(I::smodule{T}, max_length::Int)
 
 Compute a free resolution of the given module $I$ of length up to the given
 maximum length. If `max_length` is set to zero, a full length free
 resolution is computed. Each element of the resolution is itself a module.
 """
-function sres(I::smodule{T}, max_length::Int) where T <: Nemo.RingElem
+function sres(I::smodule{T}, max_length::Int) where T <: Singular.FieldElem
```
Comment (tthsqe12) [inline src/module/module.jl:196]: So now the only error is:
`MethodError: no method matching sres(::smodule{spoly{n_Q}}, ::Int64)`

This should be
`function sres(I::smodule{spoly{T}}, max_length::Int) where T <: Nemo.FieldElem`
with a
`sresolution{spoly{T}}(R, r, Bool(minimal))`
at the end.

Outside of a very limited number of cases, which I will make sure are documented soon, there isn't a reason to use `Singular.FieldElem`. It is a limitation of julia type system.

**Pull request merged**